Tribler/Core/Modules/restapi/search_endpoint.py

```diff
@@ -0,0 +1,83 @@
+import json
+import logging
+
+from twisted.web import http, resource
+from Tribler.Core.Utilities.search_utils import split_into_keywords
+from Tribler.Core.exceptions import OperationNotEnabledByConfigurationException
+from Tribler.Core.simpledefs import NTFY_CHANNELCAST, NTFY_TORRENTS, SIGNAL_TORRENT, SIGNAL_ON_SEARCH_RESULTS, \
+    SIGNAL_CHANNEL
+
+
+class SearchEndpoint(resource.Resource):
+    """
+    This endpoint is responsible for searching in channels and torrents present in the local Tribler database.
+
+    A GET request to this endpoint will create a search. Results are returned over the events endpoint. First, the
+    results available in the local database will be pushed. After that, incoming Dispersy results are pushed.
+    The query to this endpoint is passed using the url, i.e. /search?q=pioneer
+
+    Example response over the events endpoint:
+    {
```

Comment:
@whirm I agree and this is something I've been thinking about (though not implemented it). Also @brussee talked about this and he suggested about the same.

I will do a fixup for this one.